### DISCUSSION TITLE
Bunny Fonts as an alternative

### DISCUSSION THREAD

**Discussion by RVxLab** [spatie/laravel-google-fonts#25] [Ideas]
Hey there! I came across a Google Fonts alternative called [Bunny Fonts](https://fonts.bunny.net) on [an article on Hackernews](https://news.ycombinator.com/item?id=31798311) recently.

They advertise themselves as a privacy friendly drop-in replacement, which may be desirable for some people.

Seeing how similar the structure is I think it could be a nice addition to this package.

## Implementation

As Bunny Fonts provide a CSS file through a URL like Google Fonts does I think the changes required would be minor:

- `extractFontUrls` would either have its pattern extended to include `fonts.bunny.net` or have the pattern injected as a second parameter which is determined in the `fetch` method (I'm not sure if the 2nd option here would count as a BC break, I'll let you be the judge of that);
- `localiseFontUrl` would have the first parameter of `str_replace` be replaced with an array to include `https://fonts.bunny.net`

If you're interested in having this added to the package I would be more than happy to supply a PR!

Please let me know what you think.

**Comment by freekmurze** [comment]
Thanks for your suggestion.

If the code is simple, well tested and documented, we'll consider merging it in

**Comment by acharseth** [comment]
What is the status on this? We are also using Bunny Fonts (not sure if it is by Laravel default or not but I find it used in vendor/packages as well).

**Comment by bergstar** [comment]
Bump on Bunny as its much faster as well
